Demos/Blazorise.Demo/Pages/Tests/StepsPage.razor

```diff
@@ -77,7 +77,7 @@
                 <CardText>To control the navigation based on the user input you can use <Code>NavigationAllowed</Code> function.</CardText>
             </CardBody>
             <CardBody>
-                <Steps @ref="stepsRef" @bind-SelectedStep="selectedStep" NavigationAllowed="NavigationAllowed">
+                <Steps @ref="stepsRef" InitialStep="@initialStep" NavigationAllowed="NavigationAllowed">
```

Comment:
We, and users, would still want to have `@bind-SelectedStep="selectedStep"` to know which step is currently active. So I would return this part.

Reply:
As described in the proposal having two-way binding is not a solution that will lead to predictable behavior.    
Users expect that setting the param will result in its change, but that's not the case if the change is blocked by `NavigationAllowed`. Also with async method it's not even feasible. 

To know which step is currently activem one can either use the get-only `SelectedStep` property or subscribe to `SelectedStepChanged`.

Reply:
`SelectedStepChanged` changed is only raised after the `NavigationAllowed` is finished with the work. And `NavigationAllowed` should only be called within "step" methods. Two-way binding should not be blocked by `NavigationAllowed` as that will not work.

Reply:
So you want the two-way binding to bypass the `NavigationAllowed` check? That will create inconsistent behavior for different methods of moving steps... That will also change the "old" functionality, where the `NavigationAllowed` is called in two-way binding...

Reply:
My idea was to bypass the `NavigationAllowed` during two-way binding, yes. But then I went through the code and I think it can be still done by making `public string SelectedStep` as "dumb" parameter. And move all its logic from the setter into `SetParametersAsync`. The `SetParametersAsync` is asynchronous and in there we can check if the step is changed, and we can also await for any `NavigationAllowed` result. What do you think?

Reply:
Yes, this will "work," but it needs to reset the `SelectedStep` when navigation is blocked. While this isn’t technically challenging, it may lead to unexpected behavior:

- When you set `SelectedStep` (e.g., from 2 to 3) on a page, it will trigger the `NavigationAllowed` check. During this time, the UI will briefly display step 3 on the page (not in the `Steps` component), even if step 3 is blocked. This depends on how quickly the async action completes.
- Additionally, it will fire the `SelectedStepChanged` event even if the `SelectedStep` change was ultimately blocked.

This also creates a distinct behavior between these two actions:
- `SelectedStep="3"` 
- `ref.SelectStep("3")`

Is this a significant issue?

Reply:
> Is this a significant issue?

> When you set SelectedStep (e.g., from 2 to 3) on a page, it will trigger the NavigationAllowed check.

I don't think it is much of an issue. One is a declarative approach, and one is imperative. So, in declarative mode, we are forced to return to a previous state of `NavigationAllowed` returns false. The problem might be the UI flicker, but I'm not sure if it will happen at all if we prevent it during `SetParametersAsync`. Have you tried it?

> Additionally, it will fire the SelectedStepChanged event even if the SelectedStep change was ultimately blocked.

This cannot be avoided. Think of how the numeric input works. If you try to set a value below Min or above the Max, it has to retrigger ValueChanged to reset it to a valid range. This is essentially the same. We are resetting to a valid step.

---

In any case. Proceed with the `SetParametersAsync,`, and we'll see how it goes.

Reply:
the UI flicker only appears outside of the component. Not in the component itself (steps will not jump here and there)

```
This will flicker: @selectedStep

<Steps @bind-SelectedStep="selectedStep"...
```

This is because of the `SelectedStep` really being a dumb parameter which has its setter solved in the `OnParametersSet` method and only when the changes are approved by `NavigationAllowed`..